File: 3sum.py
# Note: Need to make it run faster

nums = [34,55,79,28,46,33,2,48,31,-3,84,71,52,-3,93,15,21,-43,57,-6,86,56,94,74,83,-14,28,-66,46,-49,62,-11,43,65,77,12,47,61,26,1,13,29,55,-82,76,26,15,-29,36,-29,10,-70,69,17,49]
result  = []

def threeSum(nums: list[int]):
    for number in range(len(nums)):
        for num1 in range(len(nums)):
            for num2 in range(len(nums)):
                if num1==num2 or number==num2 or number==num1:
                    continue
                else:
                    if (nums[number] + nums[num1] + nums[num2]) == 0:
                        x = [nums[number], nums[num1], nums[num2]]
                        x.sort()
                        if x not in result:
                            result.append(x)
                    else:
                        continue
    return len(result)

print(threeSum(nums))

print(len([[-82,34,48],[-49,15,34],[-70,34,36],[-43,-6,49],[-43,-6,49],[-43,-6,49],[-43,-6,49],[-43,-14,57],[-43,-3,46],[-49,-3,52],[-49,21,28],[-49,13,36],[-49,13,36],[-66,-11,77],[-43,12,31],[-14,2,12],[-49,2,47],[-3,1,2],[-49,1,48],[-66,1,65],[-49,13,36],[-14,1,13],[-49,13,36],[-49,-6,55],[-82,-11,93],[-82,21,61],[-82,26,56],[-43,15,28],[-49,13,36],[-49,13,36],[-82,36,46],[-29,-14,43],[-29,1,28],[-43,10,33],[-66,10,56],[-11,1,10],[-70,-14,84],[-70,13,57],[-70,-6,76],[-70,15,55],[-66,-3,69],[-82,13,69],[-70,1,69],[-14,-3,17],[-11,-6,17],[-82,17,65],[-43,17,26],[-29,12,17],[-43,-6,49],[-43,-6,49],[-82,33,49],[-70,21,49],[-66,17,49]]))
print(len([[-82,-11,93],[-82,13,69],[-82,17,65],[-82,21,61],[-82,26,56],[-82,33,49],[-82,34,48],[-82,36,46],[-70,-14,84],[-70,-6,76],[-70,1,69],[-70,13,57],[-70,15,55],[-70,21,49],[-70,34,36],[-66,-11,77],[-66,-3,69],[-66,1,65],[-66,10,56],[-66,17,49],[-49,-6,55],[-49,-3,52],[-49,1,48],[-49,2,47],[-49,13,36],[-49,15,34],[-49,21,28],[-43,-14,57],[-43,-6,49],[-43,-3,46],[-43,10,33],[-43,12,31],[-43,15,28],[-43,17,26],[-29,-14,43],[-29,1,28],[-29,12,17],[-14,-3,17],[-14,1,13],[-14,2,12],[-11,-6,17],[-11,1,10],[-3,1,2]]))

File: 3sum.js
const nums = [
  -12, 12, -5, -4, -12, 11, 9, -11, 13, 1, 12, -1, 8, 1, -9, -11, -13, -4, 10,
  -9, -6, -11, 1, -15, -3, 4, 0, -15, 3, 6, -4, 7, 3, -2, 10, -2, -6, 4, 2, -7,
  12, -1, 7, 6, 7, 6, 2, 10, -13, -3, 8, -12, 2, -5, -12, 6, 6, -5, 6, -5, -14,
  9, 9, -4, -8, 4, 2, -7, -15, -11, -7, 12, -4, 8, -5, -12, -1, 12, 5, 1, -5,
  -1, 5, 12, 9, 0, 3, 0, 3, -14, 2, -4, 2, -4, 0, 1, 7, -13, 9, -1, 13, -12,
  -11, -6, 11, -1, -10, -5, -3, 10, 3, 7, -6, -15, -4, 10, 1, 14, -12,
];

function threeSum(nums) {
  let result = [];
  for (let number in nums) {
    for (let number1 in nums) {
      for (let number2 in nums) {
        if (number == number2 || number1 == number2 || number1 == number) {
          continue;
        } else {
          if (nums.number + nums.number1 + nums.number2 == 0) {
            let x = [nums.number, nums.number1, nums.number2];
            if (nums.includes(x) == false) {
              result.push(x);
            }
          } else {
            continue;
          }
        }
      }
    }
  }
  return result.length;
}

console.log(threeSum(nums));
